[CDTOOL-1113] Support compression attributes in HTTPS log endpoint

mdenomy opened the pull request:
All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/fastly/go-fastly/pulls) for the same update/change?

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### New Feature Submissions:

* [x] Does your submission pass tests? 
* [x] Post the output of your test runs

### User Impact
Able to set compression for https via TF


### Are there any considerations that need to be addressed for release?
Using latest go-fastly

mdenomy committed:
Support compression attributes in HTTPS log endpoint

mdenomy committed:
Changelog

mdenomy committed:
Update docs

mdenomy committed:
Updated go-fastly dep

anthony-gomez-fastly committed:
Merge branch 'main' into mdenomy/cdtool-1113

philippschulte reviewed (changes requested):
Thank you, this looks great! I only have one small request: Please add a case to the test suite where `gzip_level = -1` to ensure it’s correctly skipped.

mdenomy committed:
Create https log with compression unspecified

Ensure that a gzip of -1 gets ignored

mdenomy committed:
Refactor attribute checks

Pull out the common checking code and pass in specific checks or data transforms

mdenomy commented:
> Thank you, this looks great! I only have one small request: Please add a case to the test suite where `gzip_level = -1` to ensure it’s correctly skipped.

@philippschulte let me know how this looks. I did some refactoring in a separate [commit](https://github.com/fastly/terraform-provider-fastly/pull/1086/commits/b5083f3c25129cd9ccacc1a0f29c5171e8c4a923) to try to share some of the common checks. Open to suggestions

philippschulte committed:
Merge branch 'main' into mdenomy/cdtool-1113

philippschulte reviewed (approved):
Great work! Thank you for your contribution @mdenomy!

Test pass:

```
TF_ACC=1 go  test $(go  list ./...) -v -run=TestAccFastlyServiceVCL_httpslogging_basic -parallel=4 -timeout 360m -ldflags="-X=github.com/fastly/terraform-provider-fastly/version.ProviderVersion=acc"
?   	github.com/fastly/terraform-provider-fastly	[no test files]
=== RUN   TestAccFastlyServiceVCL_httpslogging_basic
=== PAUSE TestAccFastlyServiceVCL_httpslogging_basic
=== RUN   TestAccFastlyServiceVCL_httpslogging_basic_compute
=== PAUSE TestAccFastlyServiceVCL_httpslogging_basic_compute
=== CONT  TestAccFastlyServiceVCL_httpslogging_basic
=== CONT  TestAccFastlyServiceVCL_httpslogging_basic_compute
--- PASS: TestAccFastlyServiceVCL_httpslogging_basic_compute (37.85s)
--- PASS: TestAccFastlyServiceVCL_httpslogging_basic (49.05s)
PASS
```

```
TF_ACC=1 go  test $(go  list ./...) -v -run=TestAccFastlyServiceVCL_httpslogging_basic_compute -parallel=4 -timeout 360m -ldflags="-X=github.com/fastly/terraform-provider-fastly/version.ProviderVersion=acc"
?   	github.com/fastly/terraform-provider-fastly	[no test files]
=== RUN   TestAccFastlyServiceVCL_httpslogging_basic_compute
=== PAUSE TestAccFastlyServiceVCL_httpslogging_basic_compute
=== CONT  TestAccFastlyServiceVCL_httpslogging_basic_compute
--- PASS: TestAccFastlyServiceVCL_httpslogging_basic_compute (35.33s)
PASS
```

```
TF_ACC=1 go  test $(go  list ./...) -v -run=TestResourceFastlyFlattenHTTPS -parallel=4 -timeout 360m -ldflags="-X=github.com/fastly/terraform-provider-fastly/version.ProviderVersion=acc"
?   	github.com/fastly/terraform-provider-fastly	[no test files]
=== RUN   TestResourceFastlyFlattenHTTPS
--- PASS: TestResourceFastlyFlattenHTTPS (0.00s)
PASS
```

anthony-gomez-fastly reviewed (approved):
LGTM, thank you for your contribution!
